waspc/cli/Command/Db/Migrate.hs

```diff
@@ -1,28 +1,90 @@
 module Command.Db.Migrate
-    ( migrate
+    ( migrateSave
+    , migrateUp
     ) where
 
+import System.IO.Error (IOError)
+import Control.Monad.Catch (catch)
 import Control.Monad.Except (throwError)
 import Control.Monad.IO.Class (liftIO)
+import qualified Path as P
+import Path.IO
 
-import StrongPath ((</>))
+import StrongPath ((</>), Abs, Dir, Path)
+import qualified StrongPath as SP
 import Command (Command, CommandError(..))
 import Command.Common (findWaspProjectRootDirFromCwd, waspSaysC)
 import qualified Common
+import Common (WaspProjectDir)
+
+-- Wasp generator interface.
 import qualified Generator.DbGenerator.Operations as DbOps
+import Generator.DbGenerator (dbRootDirInProjectRootDir)
+import Generator.Common (ProjectRootDir)
 
 
-migrate :: String -> Command ()
-migrate migrationName = do
+migrateSave :: String -> Command ()
+migrateSave migrationName = do
     waspRootDir <- findWaspProjectRootDirFromCwd
     let genProjectRootDir = waspRootDir </> Common.dotWaspDirInWaspProjectDir </>
-                         Common.generatedCodeDirInDotWaspDir
+                            Common.generatedCodeDirInDotWaspDir
 
-    waspSaysC "Saving db migration..."
+    waspSaysC "Checking for changes in schema to save..."
     migrateSaveResult <- liftIO $ DbOps.migrateSave genProjectRootDir migrationName
     case migrateSaveResult of
-        Left migrateSaveError -> throwError $ CommandError $ "Migrate save failed: " ++ migrateSaveError
-        Right () -> waspSaysC "Migration has been successfully saved."
+        Left migrateSaveError -> throwError $ CommandError $ "Migrate save failed: " ++
+                                 migrateSaveError
+        Right () -> waspSaysC "Done."
+
+    waspSaysC "Copying migrations folder from Prisma to Wasp project..."
+    copyDbMigDirResult <- liftIO $ copyDbMigrationsDir waspRootDir genProjectRootDir
+
+    case copyDbMigDirResult of
+        Nothing -> waspSaysC "Done."
+        Just err -> throwError $ CommandError $ "Copying migration folder failed: " ++ err
+
+    applyAvailableMigrationsAndGenerateClient genProjectRootDir
 
     waspSaysC "All done!"
 
+    where
+        copyDbMigrationsDir
+            :: Path Abs (Dir WaspProjectDir)
```

Comment:
Ah uff which is which :D. I really named these badly. Maybe name them in a qualified manner so it is clear which one is coming from the generator? We really should update all of these names.